packages/wallet/src/redux/direct-funding/depositing/reducer.ts

```diff
@@ -1,29 +1,49 @@
-import { unreachable } from '../../../../utils/reducer-utils';
-import { createDepositTransaction } from '../../../../utils/transaction-generator';
-import { StateWithSideEffects } from '../../../utils';
+import { unreachable } from '../../../utils/reducer-utils';
+import { createDepositTransaction } from '../../../utils/transaction-generator';
+import { StateWithSideEffects } from '../../utils';
 
-import * as actions from '../../../actions';
+import * as actions from '../../actions';
 import * as states from './state';
-import * as fundingStates from '../../direct-funding-state/state';
-import { WalletProcedure } from '../../../types';
+import * as fundingStates from '../state';
+import { WalletProcedure } from '../../types';
+import { ProtocolReducer, ProtocolStateWithSharedData } from '../../protocols';
+import { SideEffects } from '../../outbox/state';
+import { accumulateSideEffects } from '../../outbox';
 
-export const depositingReducer = (
-  state: states.Depositing,
+export const depositingReducer: ProtocolReducer<fundingStates.DirectFundingState> = (
+  state: ProtocolStateWithSharedData<states.Depositing>,
   action: actions.WalletAction,
-): StateWithSideEffects<fundingStates.DirectFundingState> => {
-  switch (state.depositStatus) {
+): ProtocolStateWithSharedData<fundingStates.DirectFundingState> => {
+  const { protocolState } = state;
+
+  switch (protocolState.depositStatus) {
     case states.WAIT_FOR_TRANSACTION_SENT:
-      return waitForTransactionSentReducer(state, action);
+      return applyUpdate(state, waitForTransactionSentReducer(protocolState, action));
     case states.WAIT_FOR_DEPOSIT_APPROVAL:
-      return waitForDepositApprovalReducer(state, action);
+      return applyUpdate(state, waitForDepositApprovalReducer(protocolState, action));
     case states.WAIT_FOR_DEPOSIT_CONFIRMATION:
-      return waitForDepositConfirmationReducer(state, action);
+      return applyUpdate(state, waitForDepositConfirmationReducer(protocolState, action));
     case states.DEPOSIT_TRANSACTION_FAILED:
-      return depositTransactionFailedReducer(state, action);
+      return applyUpdate(state, depositTransactionFailedReducer(protocolState, action));
   }
-  return unreachable(state);
+  return unreachable(protocolState);
 };
 
+function applyUpdate(
```

Comment:
I think it'd be nice to change the subReducers to take a `ProtocolStateWithSharedData` so we don't need this helper function and makes the reducer easier to understand.

Longer term that will allow us to `StateWithSideEffects` and hopefully simplify the `SideEffects`